File: NATSInternal.Web/Models/Shared/PhotoBasicModel.cs
using System.ComponentModel;
using NATSInternal.Application.Localization;
using NATSInternal.Application.UseCases.Shared;

namespace NATSInternal.Web.Models.Shared;

public class PhotoBasicModel
{
    #region Constructors
    public PhotoBasicModel(PhotoBasicResponseDto responseDto)
    {
        Id = responseDto.Id;
        Url = responseDto.Url;
        IsThumbnail = responseDto.IsThumbnail;
    }
    #endregion
    
    #region Properties
    public Guid Id { get; }
    public string Url { get; }
    public bool IsThumbnail { get; }
    #endregion
}

File: NATSInternal.Frontend/NATSInternal.React/src/models/shared/photoBasicModel.ts
declare global {
  type PhotoBasicModel = Readonly<{
    id: number;
    url: string;
    isThumbnail: boolean;
  }>;
}

export function createPhotoBasicModel(responseDto: PhotoBasicResponseDto): PhotoBasicModel {
  return {
    id: responseDto.id,
    url: responseDto.url,
    isThumbnail: responseDto.isThumbnail
  };
}
